packages/ra-supabase-core/src/authProvider.ts

```diff
@@ -139,7 +139,22 @@ export const supabaseAuthProvider = (
             return Promise.resolve();
         },
         async getPermissions() {
-            return;
+            const {
+                data,
+                error,
+            } = await client.auth.getUser()
+            if (error) {
+                throw error;
+            }
+            if (data.user == null) {
+                throw new Error();
+            }
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏If the user is null (e.g. because not yet signed in), we should not reject here, because it will raise an error in the console. We simply need to return an empty array, or undefined, in that case. (It is not the responbisbility of the `getPermissions` function to check whether the user is signed in or not, this is the role of the `checkAuth` function)

```suggestion
            if (data.user == null) {
                return undefined;
            }
```

Reply:
@slax57 given that should be doing the same on getUser?

Reply:
We could, but my understanding is that this would be a 'real' error we would want to log, not simply an error due to the fact that the user is not yet signed in. So to me it also makes sense to rethrow the error in that case.

Reply:
Cool, I'll leave that alone for now though it should probably rethrow the existing error and only make a new one if it's not present.